templates/api/rpc/client.go.tpl

```diff
@@ -0,0 +1,118 @@
+	"github.com/getoutreach/services/pkg/grpcx"
+	"github.com/getoutreach/{{ .Config.Name }}/api"
+
+	{{- $additionalImports := stencil.GetModuleHook "api/rpc/client.additionalImports" }}
+	{{- if $additionalImports }}
+	// imports added by modules
+		{{- range $additionalImports }}
+	{{ . | quote }}
+		{{- end }}
+	// end imports added by modules
+	{{- end }}
+)
+
+// New returns a new grpc client for the {{ .Config.Name }} Service
+//
+// The client is concurrency safe and handles reconnecting.
+// All calls automatically handle logging, tracing, metrics,
+// service discovery, and authn.
+func New(ctx context.Context) (api.Service, error) {	
+	{{- $initializeClient := stencil.GetModuleHook "api/rpc/client.initializeClient" }}
+	{{- if $initializeClient }}
+	// Inserted by modules
+	{{- range $initializeClient }}
+	{{ . }}
+	{{- end }}
+	// End Inserted by modules
+	{{- end }}
+
+	clientOpts := []grpcx.ClientConnOption{
+		grpcx.WithServiceDiscovery(),
+
+		{{- $clientOpts := stencil.GetModuleHook "api/rpc/client.clientOpts" }}
+		{{- if $clientOpts }}
+		// Inserted by modules
+		{{- range $clientOpts }}
+		{{ . }},
+		{{- end }}
+		// End Inserted by modules
+		{{- end }}
+	}
+
+	conn, err := grpcx.NewClientConn(ctx, "{{ .Config.Name }}", useAuthn, useDiscovery)
+	if err != nil {
+		return nil, err
+	}
+	return &client{
+		closers: []func(ctx context.Context) error{
+			{{- $closers := stencil.GetModuleHook "api/rpc/client.closers" }}
+			{{- if $closers }}
+			// Closers Inserted by modules
+			{{- range $closers }}
+			{{ . }},
+			{{- end }}
+			// End closers inserted by modules
+			{{- end }}
+			func(_ context.Context) error { return conn.Close() },
+		},
+	  {{ title $pkgName }}Client: api.New{{ title $pkgName }}Client(conn),
+	}, nil
+}
+
+// client is the type that actually implements the correct interface to serve as
+// a gRPC client for the rms service as per the protobuf files.
+type client struct {
+	closers []func(ctx context.Context) error
+	api.{{ title $pkgName }}Client
+	// Place your client struct data here
+}
+
+// Close is necessary to avoid potential resource leaks
+func (c client) Close(ctx context.Context) error {
+	errors := make([]error, 0)
+	for _, fn := range c.closers {
+		if err := fn(ctx); err != nil {
+			errors = append(errors, err)
+		}
+	}
+	if len(errors) != 0 {
+		return fmt.Errorf("failed to close client: %v", errors)
+	}
+
+	return nil
+}
+
+// Place any client handler functions for your service here
+func (c client) Ping(ctx context.Context, message string) (string, error) {
+	in := &api.PingRequest{Message: message}
+	resp, err := c.{{ title $pkgName }}Client.Ping(ctx, in)
+	if err != nil {
+		return "", err
+	}
+	return resp.Message, nil
+}
+
+func (c client) Pong(ctx context.Context, message string) (string, error) {
+	in := &api.PongRequest{Message: message}
+	resp, err := c.{{ title $pkgName }}Client.Pong(ctx, in)
+	if err != nil {
+		return "", err
+	}
+	return resp.Message, nil
+}
```

Comment:
Are these endpoints used for anything in the system, or do they just serve as examples?

Reply:
These are copied, verbatim, from `bootstrap` for the default endpoints generated by bootstrap

Reply:
Yes, they're sourced from `bootstrap` - but are they relied on by our infrastructure at all (like as health check endpoints)?

Reply:
Ah, I see what you're saying now. Nope, just example endpoints with no value :D

Reply:
I've filed a ticket to put these in the `block` sections (new files only) so that users can delete them. It's very silly to have these around forever!

Reply:
Ah, so I actually missed it but this is a static file, so it doesn't need blocks :D